Classes/DataProcessing/DataProcessingTrait.php

```diff
@@ -20,18 +20,33 @@ trait DataProcessingTrait
      */
     protected function removeDataIfnotAppendInConfiguration(array $processorConfiguration, array $processedData): array
     {
-        if (!isset($processorConfiguration['appendData']) ||
-            (int)$processorConfiguration['appendData'] === 0) {
-            unset($processedData['data']);
-            if (isset($processorConfiguration['as'], $processedData[$processorConfiguration['as']])
-                && is_array($processedData[$processorConfiguration['as']])) {
+        if (
+            !isset($processorConfiguration['appendData']) ||
+            (int)$processorConfiguration['appendData'] !== 1
+        ) {
+            // Items to keep
+            $removeAll = !isset($processorConfiguration['appendData']) || $processorConfiguration['appendData'] == 0;
+            $keepItems = $removeAll ? [] : array_flip(array_map('trim', explode(',', $processorConfiguration['appendData'])));
```

Comment:
I think we should add new option like `keep(Items|fields)`  or `passFields` to keep clear what is happening also do not change behavior of flag `appendData` to keep BC, also you can use `GeneralUtility:trimExplode`